comit/src/network/swap_digest.rs

```diff
@@ -1,19 +1,42 @@
 use crate::{asset, identity, network::protocols::announce::SwapDigest, RelativeTime, Timestamp};
-use digest::Digest;
+use digest::{Digest, ToDigestInput};
+
+// TODO should we use swap_protocols::SwapProtocol instead?
+#[derive(Clone, Digest, Debug, PartialEq, Copy)]
+#[digest(hash = "SwapDigest")]
+pub enum SwapProtocol {
+    #[digest(prefix = "4b17")]
+    Hbit,
+    #[digest(prefix = "e5ec")]
+    Herc20,
+    #[digest(prefix = "c3d3")]
+    Halight,
+}
+
+// TODO why is this necessary?
```

Comment:
Because you assigned a prefix to a `SwapProtocol` field line 87 of facade.rs.

Also, we initially decided against the support of nested structs for digest with @thomaseizinger so I guess you hacked around that.

I would strongly suggest to update the digest crate to implement and document a streamlined way to support nested structs.

Reply:
> I would strongly suggest to update the digest crate to implement and document a streamlined way to support nested structs.

I agree with this but I would like us to first try and use the current version so we get a better grasp of what we actually need.
This usecase for example hints that we might need a 2nd derive for `ToDigestInput` that works on enums.

Reply:
Hum, it already works on enums. Not sure what is missing.

Reply:
To my knowledge, we are only able to derive `Digest`? `ToDigestInput` needs to be implemented manually, no?